addons/hr_work_entry_holidays/tests/test_leave.py

```diff
@@ -199,3 +199,64 @@ def test_time_off_duration_contract_state_change(self):
         self.assertTrue(leave.number_of_days, 1)
         contract.state = "open"
         self.assertTrue(leave.number_of_days, 1)
+
+    def test_split_leaves_by_entry_type(self):
+        employee = self.env['hr.employee'].create({'name': 'Test'})
+
+        self.env['hr.contract'].create({
+            'employee_id': employee.id,
+            'resource_calendar_id': self.calendar_40h.id,
+            'date_start': datetime(2024, 9, 1),
+            'date_end': datetime(2024, 9, 30),
+            'name': 'Contract',
+            'wage': 5000.0,
+            'state': 'open',
+        })
```

Comment:
```suggestion
        employee = self.richard_emp
```

Is there a reason you can't re-use the contracted employee that already exists for the test suite? Creating new records for each individual test adds unnecessary overhead, so should be avoided when possible.

Reply:
I've tried Richard but he doesn't have a contract (which is needed to generate work entries). There is another one which I might be able to use though, I'll try

Reply:
He does, but it's in draft, so you can just
```python
self.richard_emp.contract_ids[0].state = 'open'
```
Or yeah, seems like `self.jules_emp` already has an open contract, so that should also work.